src/Native/Unix/System.Native/pal_interfaceaddresses.c

```diff
@@ -146,56 +146,55 @@ extern "C" int32_t SystemNative_EnumerateInterfaceAddresses(IPv4AddressFound onI
 }
 
 #if HAVE_RT_MSGHDR
-extern "C" int32_t SystemNative_EnumerateGatewayAddressesForInterface(uint32_t interfaceIndex, GatewayAddressFound onGatewayFound)
+int32_t SystemNative_EnumerateGatewayAddressesForInterface(uint32_t interfaceIndex, GatewayAddressFound onGatewayFound)
 {
     int routeDumpName[] = {CTL_NET, AF_ROUTE, 0, AF_INET, NET_RT_DUMP, 0};
 
     size_t byteCount;
 
-    if (sysctl(routeDumpName, 6, nullptr, &byteCount, nullptr, 0) != 0)
+    if (sysctl(routeDumpName, 6, NULL, &byteCount, NULL, 0) != 0)
     {
         return -1;
     }
 
-    uint8_t* buffer = new (std::nothrow) uint8_t[byteCount];
-    if (buffer == nullptr)
+    uint8_t* buffer = malloc(byteCount);
+    if (buffer == NULL)
     {
         errno = ENOMEM;
         return -1;
     }
 
-    while (sysctl(routeDumpName, 6, buffer, &byteCount, nullptr, 0) != 0)
+    while (sysctl(routeDumpName, 6, buffer, &byteCount, NULL, 0) != 0)
     {
-        delete[] buffer;
-        buffer = new (std::nothrow) uint8_t[byteCount];
-        if (buffer == nullptr)
+        buffer = realloc (buffer, byteCount);
```

Comment:
cast from void pointer to uint8_t pointer, or make it a void pointer if that works